### PR TITLE
Cypress tutorial

### PR DESCRIPTION
**Pull request by jakubnowicki**
### Changes
Adds Cypress tutorial

### How to test
Run `pkgdown::build_site()` to check the version with the new tutorial.

### CODE REVIEW

File: vignettes/tutorial/cypress.Rmd [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+
+![](images/clicks_e2e_test_1.png)
+
+## Test if clicks are counted correctly
+
+The next step is to check if clicking the button affects the counter
+and if it shows the correct value. To do that, we will simulate 5 clicks
+and check if the counter is set to 5.
+
+Since the button will be clicked several times, we need to assign it to a variable.
```
Comment (kamilzyla) [inline vignettes/tutorial/cypress.Rmd:411]: Technically it's called an alias; [variables can work in somewhat unexpected way](https://docs.cypress.io/guides/core-concepts/variables-and-aliases).

**Review by kamilzyla — APPROVED** [review]
Nice tutorial, thanks! Everything works :grin: :+1: 

Two additional comments:
1. We have a "Tutorial" link in the README. Perhaps we should just replace it with "Tutorials" text.
2. A couple of code examples have both a title and a comment at the top (looks repetitive), e.g. `app/view/message.R`. Perhaps stick to code comments for file names?

File: vignettes/tutorial/cypress.Rmd [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+Since we have just one simple test, there will be not much there, but soon this will change.
+
+# Test the message functionality message
+
+We are about to write our first end-to-end test. The goal is to check if
+clicking the button with "Show message" label will show the message.
+
+## Test file structure
+
+The first thing to do is to create a test file: `tests/cypress/integration/message.js`.
```
[the remark this answers is not in the document]
Comment (jakubnowicki) [reply vignettes/tutorial/cypress.Rmd:181]: 1. I'll change the tutorial (I don't want to introduce changes in the code).
2. Existing file is a valid test (app is/is not able to start) and a place to write more advanced "starting" tests (app starts and does not crash, etc)

**Comment by jakubnowicki** [comment]
> Nice tutorial, thanks! Everything works grin +1
> 
> Two additional comments:
> 
> 1. We have a "Tutorial" link in the README. Perhaps we should just replace it with "Tutorials" text.
> 2. A couple of code examples have both a title and a comment at the top (looks repetitive), e.g. `app/view/message.R`. Perhaps stick to code comments for file names?

1. I think that the README should link to the basic tutorial - the first place to go when you start learning Rhino.
2. Makes sense, will change that.

**Pull request merged**